### PR TITLE
scrcpy: 1.24 -> 1.25

### PR DESCRIPTION
**Pull request by msfjarvis**
###### Description of changes

https://github.com/Genymobile/scrcpy/releases/tag/v1.25

Notably this release brings compatibility with Android 13 and adds desktop entries.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SubhrajyotiSen** [comment]
Result of `nixpkgs-review pr 207242` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>scrcpy</li>
  </ul>
</details>

**Review by SubhrajyotiSen — APPROVED** [review]

**Comment by markuskowa** [comment]
This breaks the `x86_64-darwin` build?

**Comment by msfjarvis** [comment]
> This breaks the `x86_64-darwin` build?

Should be fixed

File: pkgs/misc/scrcpy/default.nix [hunk 1]
```diff
@@ -52,6 +53,9 @@ stdenv.mkDerivation rec {
 
     # runtime dep on `adb` to push the server
     wrapProgram "$out/bin/scrcpy" --prefix PATH : "${platform-tools}/bin"
+  '' + lib.optionals stdenv.isLinux ''
```
Comment (markuskowa) [inline pkgs/misc/scrcpy/default.nix:56]:
```suggestion
  '' + lib.optionalString stdenv.isLinux ''
```

File: pkgs/misc/scrcpy/default.nix [hunk 2]
```diff
@@ -52,6 +53,9 @@ stdenv.mkDerivation rec {
...
+    substituteInPlace $out/share/applications/scrcpy-console.desktop \
+      --replace "/bin/bash" "${bash}/bin/bash"
```
Comment (markuskowa) [inline pkgs/misc/scrcpy/default.nix:58]:
```suggestion
      --replace "/bin/bash" "${runtimeShell}"
```

**Review by markuskowa — CHANGES_REQUESTED** [review]

**Review by markuskowa — APPROVED** [review]

**Pull request merged**